src/builtin.cpp

```diff
@@ -2137,9 +2138,10 @@ static int builtin_read(parser_t &parser, io_streams_t &streams, wchar_t **argv)
     int nchars = 0;
     bool shell = false;
     bool array = false;
+    bool silent = false;
     bool split_null = false;
 
-    const wchar_t *short_options = L"ac:ghlm:n:p:suxzP:UR:";
+    const wchar_t *short_options = L"ac:ghlm:n:p:siuxzP:UR:";
```

Comment:
Notice that we are trying to keep the flags sorted so adding the `i` between the `hl` pair is better.

P.S., Yes, there is lots of code elsewhere in fish that violates this best practice but Rome wasn't built in a day.